components/DonationListItem.js

```diff
@@ -0,0 +1,88 @@
+import styled from "styled-components";
+import { useState } from "react";
+
+export default function DonationListItem({ donation }) {
+  const [toggle, setToggle] = useState(false);
+  function handleClick() {
+    setToggle(!toggle);
+  }
+  <h2> Spenden in deiner Nähe: </h2>;
+  if (toggle) {
+    return (
+      <StyledCard>
+        <p>{donation.image}</p>
+        <p>Kategorie: {donation.category}</p>
+        <p>Beschreibung: {donation.description}</p>
+        <p>Entfernung: {donation.distance}</p>
+        <p>Haltbar bis: {donation.good_before}</p>
+        <p>Kontakt: {donation.contact_information}</p>
+        <label for="checkbox1">Bio</label>
+        <input
+          type="checkbox"
+          id="checkbox1"
+          name="checkbox1"
+          value="value1"
+          checked={donation.isBio}
+        />
+        <label for="checkbox2">Vegetarisch</label>
+        <input
+          type="checkbox"
+          id="checkbox2"
+          name="checkbox2"
+          value="value2"
+          checked={donation.isVegetarian}
+        />
+        <label for="checkbox3">Vegan</label>
+        <input
+          type="checkbox"
+          id="checkbox3"
+          name="checkbox3"
+          value="value3"
+          checked={donation.isVegan}
+        />
+        <label for="checkbox4">Glutenfrei</label>
+        <input
+          type="checkbox"
+          id="checkbox4"
+          name="checkbox4"
+          value="value4"
+          checked={donation.isGlutenfree}
+        />
+        <StyledLessInfoButton type="button" onClick={() => handleClick()}>
+          Weniger Infos
+        </StyledLessInfoButton>
+      </StyledCard>
+    );
+  } else {
+    return (
+      <StyledCard>
+        <p>{donation.image}</p>
+        <p>Beschreibung: {donation.description}</p>
+        <p>Entfernung: {donation.distance}</p>
+        <StyledMoreInfoButton type="button" onClick={() => handleClick()}>
+          Mehr Infos
+        </StyledMoreInfoButton>
+      </StyledCard>
+    );
```

Comment:
I like your approach a lot with the local state. Your conditional rendering is fine, but be aware that you could solve with by using the `&&` operator. If you don't know how to do that I would recommend you to research it. But again, very nice solution here!